src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,272 @@
+            {
+                return true;
+            }
+
+            // explicit conversion cases: 
+            //      (type)expr, expr is type, expr as type
+            if (initializerExpression.IsKind(SyntaxKind.CastExpression) ||
+                initializerExpression.IsKind(SyntaxKind.IsExpression) ||
+                initializerExpression.IsKind(SyntaxKind.AsExpression))
+            {
+                return true;
+            }
+
+            // other Conversion cases:
+            //      a. conversion with helpers like: int.Parse, TextSpan.From methods 
+            //      b. types that implement IConvertible and then invoking .ToType()
+            //      c. System.Convert.Totype()
+            var declaredTypeSymbol = semanticModel.GetTypeInfo(variableDeclaration.Type, cancellationToken).Type;
+            var expressionOnRightSide = initializerExpression.WalkDownParentheses();
+
+            var memberName = expressionOnRightSide.GetRightmostName();
+            if (memberName == null)
+            {
+                return false;
+            }
+
+            var possibleMethodSymbol = semanticModel.GetSymbolInfo(memberName, cancellationToken).Symbol;
+            if (possibleMethodSymbol == null || !possibleMethodSymbol.IsKind(SymbolKind.Method))
+            {
+                return false;
+            }
+
+            var methodSymbol = (IMethodSymbol)possibleMethodSymbol;
+            if (memberName.IsRightSideOfDot())
+            {
+                var typeName = memberName.GetLeftSideOfDot();
+                return PossibleConversionMethod(methodSymbol, declaredTypeSymbol, semanticModel, typeName, cancellationToken);
+            }
+
+            return false;
+        }
+
+        protected bool IsIntrinsicType(SyntaxNode declarationStatement) =>
+            declarationStatement.IsKind(SyntaxKind.VariableDeclaration)
+            ? ((VariableDeclarationSyntax)declarationStatement).Variables.Single().Initializer.Value.IsAnyLiteralExpression()
+            : false;
+
+        private ExpressionSyntax GetInitializerExpression(EqualsValueClauseSyntax initializer) =>
+            initializer.Value is CheckedExpressionSyntax
+                ? ((CheckedExpressionSyntax)initializer.Value).Expression
+                : initializer.Value;
+
+        private bool PossibleConversionMethod(IMethodSymbol methodSymbol, ITypeSymbol declaredType, SemanticModel semanticModel, ExpressionSyntax typeName, CancellationToken cancellationToken)
+        {
+            if (methodSymbol.ReturnsVoid)
+            {
+                return false;
+            }
+
+            var typeInInvocation = semanticModel.GetTypeInfo(typeName, cancellationToken).Type;
+
+            // case: int.Parse or TextSpan.FromBounds
+            if (methodSymbol.Name.StartsWith("Parse", StringComparison.Ordinal) 
+                || methodSymbol.Name.StartsWith("From", StringComparison.Ordinal))
```

Comment:
These are probably too general. I can easily imagine false positives (especially for "From"). If there are only a handful of methods we care about in the BCL and in Roslyn, why not just specifically look for those?